saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/OpenSaml4AuthenticationProvider.java

```diff
@@ -404,6 +412,26 @@ public static Converter<ResponseToken, Saml2ResponseValidatorResult> createDefau
 		};
 	}
 
+	private static String getStatusCode(Response response) {
+		if (response.getStatus() == null) {
+			return StatusCode.SUCCESS;
+		}
+		if (response.getStatus().getStatusCode() == null) {
+			return StatusCode.SUCCESS;
+		}
+
+		StatusCode parentStatusCode = response.getStatus().getStatusCode();
+		String parentStatusCodeValue = parentStatusCode.getValue();
+		if (includeChildStatusCodes.contains(parentStatusCodeValue)) {
+			return Optional.ofNullable(parentStatusCode.getStatusCode())
+					.map(StatusCode::getValue)
+					.map(childStatusCodeValue -> parentStatusCodeValue + childStatusCodeValue)
```

Comment:
I think concatenating the messages is something more of a concern for the view. Would you please change `getStatusCode` to something like `getStatusCodes` and return a list of strings? Then the result object should add a `Saml2Error` for each one.